server/consumer.go

```diff
@@ -2247,16 +2247,21 @@ func (o *consumer) checkPendingRequests() {
 // This will release any pending pull requests if applicable.
 // Should be called only by the leader being deleted or stopped.
 // Lock should be held.
-func (o *consumer) releaseAnyPendingRequests() {
+func (o *consumer) releaseAnyPendingRequests(isConsumerAssigned bool) {
 	if o.mset == nil || o.outq == nil || o.waiting.len() == 0 {
 		return
 	}
-	hdr := []byte("NATS/1.0 409 Consumer Deleted\r\n\r\n")
+	var hdr []byte
+	if !isConsumerAssigned {
+		hdr = []byte("NATS/1.0 409 Consumer Deleted\r\n\r\n")
```

Comment:
If we remove a pending request, we must inform the application. We just need to change the phrasing of the advisory.